Validation/RecoTrack/plugins/TrackingNtuple.cc

```diff
@@ -2660,6 +2667,10 @@ void TrackingNtuple::fillSimHits(const TrackerGeometry& tracker,
     simhit_x.push_back(pos.x());
     simhit_y.push_back(pos.y());
     simhit_z.push_back(pos.z());
+    if (saveModuleType_) {
+      const auto moduleType = tracker.getDetectorType(detId);
+      simhit_moduleType.push_back(moduleType);
```

Comment:
I was also thinking that a better way could be just to add the type in the same part as the DetId details https://github.com/trackreco/cmssw/blob/2ca698a1a211353b9681326d6c227c7da9e189f4/Validation/RecoTrack/plugins/TrackingNtuple.cc#L694-L706

Reply:
Codewise, it would make sense to add it here but then we would have to add it to multiple other collections apart from the `simhit` and `ph2` ones.

To include this in the `ph2` collection, a few more places in the code need to be modified. I didn't do it for now, waiting for more info from Philip to understand how we can include this information in the Segment Linking setup. If you think it's worth it in any case, I can do it.